src/backend/distributed/commands/multi_copy.c

```diff
@@ -2828,6 +2831,25 @@ CopyStatementHasFormat(CopyStmt *copyStatement, char *formatName)
 }
 
 
+#if PG_VERSION_NUM >= PG_VERSION_15
+
+/*
+ * ErrorIfMergeInCopy Raises an exception if the MERGE is called in the COPY.
+ */
+static void
+ErrorIfMergeInCopy(CopyStmt *copyStatement)
+{
+	if (!copyStatement->relation && (IsA(copyStatement->query, MergeStmt)))
+	{
+		ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+						errmsg("MERGE not supported in COPY")));
+	}
+}
+
+
+#endif
```

Comment:
Style suggestion: put PG_VERSION_NUM check inside the ErrorIfMergeInCopy() function body, like GetMergeJoinConditionList(). Then the PG_VERSION_NUM check on line 2863 can be removed.

Reply:
Done.